src/config/navigation/tabConfig.tsx

```diff
@@ -142,17 +143,29 @@ export const createDynamicTabConfig = (
     tabBarAccessibilityLabel: `${accessibilityLabel || label} (Tab ${index + 1} von ${totalCount})`,
     tabBarLabel: label,
     tabBarLabelStyle,
-    tabBarIcon: ({ color }: TabBarIconProps) => (
-      <OrientationAwareIcon
-        color={color}
-        Icon={Icon[iconName as keyof typeof Icon]}
-        iconName={iconName}
-        landscapeStyle={iconLandscapeStyle}
-        size={normalize(iconSize)}
-        strokeColor={strokeColor}
-        strokeWidth={strokeWidth}
-        style={iconStyle}
-      />
-    )
+    tabBarIcon: ({ color, focused }: TabBarIconProps) =>
+      !!activeIconName && focused ? (
+        <OrientationAwareIcon
+          color={color}
+          Icon={Icon[activeIconName as keyof typeof Icon]}
+          iconName={activeIconName}
+          landscapeStyle={iconLandscapeStyle}
+          size={normalize(iconSize)}
+          strokeColor={strokeColor}
+          strokeWidth={strokeWidth}
+          style={iconStyle}
+        />
+      ) : (
+        <OrientationAwareIcon
+          color={color}
+          Icon={Icon[iconName as keyof typeof Icon]}
+          iconName={iconName}
+          landscapeStyle={iconLandscapeStyle}
+          size={normalize(iconSize)}
+          strokeColor={strokeColor}
+          strokeWidth={strokeWidth}
+          style={iconStyle}
+        />
```

Comment:
too much code duplicated here sadly. isn't it possible to set the icon name conditionally and then return the `OrientationAwareIcon` once?

Reply:
Yes, you are right. I prevented code repetition with the commit I just sent. I added !`!activeIconName && focused` control to `iconName` and `Icon` props.

https://github.com/smart-village-solutions/smart-village-app-app/commit/f04288440c71c50bbfec9e3945ba714d4baf4e0f

Reply:
wow, such great small changes now!